phpBB/phpbb/template/twig/node/event.php

```diff
@@ -42,40 +46,80 @@ public function compile(\Twig\Compiler $compiler)
 
 		$location = $this->listener_directory . $this->getNode('expr')->getAttribute('name');
 
+		$compiler_steps = [];
+
 		foreach ($this->environment->get_phpbb_extensions() as $ext_namespace => $ext_path)
 		{
 			$ext_namespace = str_replace('/', '_', $ext_namespace);
 
+			$compiler_calls = [];
+
 			if ($this->environment->isDebug())
 			{
 				// If debug mode is enabled, lets check for new/removed EVENT
 				//  templates on page load rather than at compile. This is
 				//  slower, but makes developing extensions easier (no need to
 				//  purge the cache when a new event template file is added)
-				$compiler
-					->write("if (\$this->env->getLoader()->exists('@{$ext_namespace}/{$location}.html')) {\n")
-					->indent()
-				;
+				$compiler_calls[] = function() use($compiler, $ext_namespace, $location) {
```

Comment:
This call always should be the 1st one, hence always at the top of "krsorted" array.